src/qilib/data_set/mongo_data_set_io.py

```diff
@@ -0,0 +1,200 @@
+"""Quantum Inspire library
+
+Copyright 2019 QuTech Delft
+
+qilib is available under the [MIT open-source license](https://opensource.org/licenses/MIT):
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import base64
+from typing import Optional, Dict, Any, Union, Tuple
+
+import numpy as np
+from bson import ObjectId
+from pymongo import MongoClient
+from pymongo.change_stream import CollectionChangeStream
+from pymongo.errors import DuplicateKeyError
+
+from qilib.data_set import DataArray
+
+
+class DocumentNotFoundError(Exception):
+    """ Error that is raised when document is not found."""
+
+
+class FieldNotUniqueError(Exception):
+    """ Error raised if field is not unique"""
+
+
+class NumpyKeys:
+    """The custom values types for encoding and decoding numpy arrays."""
+    OBJECT: str = '__object__'
+    CONTENT: str = '__content__'
+    DATA_TYPE: str = '__data_type__'
+    SHAPE: str = '__shape__'
+    ARRAY: str = '__ndarray__'
+
+
+class MongoDataSetIO:
+    """ Helper class for the MongoDataSetIOReader and -Writer"""
+
+    DEFAULT_DATABASE = 'qilib'
```

Comment:
Maybe a bit anal, but this is not a database, it's a database name:

```suggestion
    DEFAULT_DATABASE_NAME = 'qilib'
```